instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/HandlerAdapterInstrumentation.java

```diff
@@ -64,10 +64,10 @@ public static void nameResourceAndStartSpan(
       if (serverSpan != null) {
         // Name the parent span based on the matching pattern
         tracer().onRequest(context, serverSpan, request);
-        // Now create a span for handler/controller execution.
-        span = tracer().startHandlerSpan(handler);
-        scope = context.with(span).makeCurrent();
       }
+      // Now create a span for handler/controller execution.
+      span = tracer().startHandlerSpan(handler);
+      scope = context.with(span).makeCurrent();
     }
```

Comment:
is this change related to this PR? if not, can you send as a separate PR? since it's a behavior change, just want it to be more visible for others to review

Reply:
Appserver tests only have a servlet, there the span name is set in servlet instrumentation and does not use `ServletContextPath.prepend`. `SpringBootSmokeTest` could possibly fail (din't really try), but it uses an empty context path so `ServletContextPath.prepend` doesn't do anything there anyway.

Probably should have mentioned this change as it should be the only non test change that was needed and there is a good chance that this is not the correct way to handle this case. This is needed for spring mvc `SpringBootBasedTest "test spans with auth error"` and a coupe of other spring mvc tests where the common theme is that these tests use an error page. This tests sets status to `401 Unauthorized` after which servlet code exits and closes server span, after that tomcat forwards request to error page with request dispatcher. This error page is also implemented with spring so it hits servlet integration again where it is handled in `Servlet3HttpServerTracer.needsRescoping(attachedContext)` branch that restores the scope of the original request. Now when code reaches `ControllerAdvice` `BaseTracer.getCurrentServerSpan(context);` returns the already ended server span that was created in first servlet call and all is fine.
When servlet common instrumentation is enable then `RequestDispatcherAdvice` sets up a new scope, this makes servlet integration not take the `Servlet3HttpServerTracer.needsRescoping(attachedContext)` branch and eventually when the request reaches `ControllerAdvice` `BaseTracer.getCurrentServerSpan(context);` is null and creating handler span is skipped.

Reply:
I reverted this changed and instead made springmvc tests run with tomcat instrumentation. With tomcat instrumentation `RequestDispatcherAdvice` does not cause loosing server span. Also fixed peer port and peer ip reporting in `TomcatTracer`. There is a slight side effect from using tomcat instrumentation, perviously showing error page did not update server span name as server span had already ended, now with tomcat instrumentation server span has not ended and error page updates server span name to `/error`.